#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Fix use of a GNU extension in inline math functions.
# DP: Related bugs: #207221, #208016.
# DP: Author: Thomas Richter <thor@math.TU-Berlin.DE>
# DP: Upstream status: Pending
# DP: Status Details: Submitting
# DP: Date: 2003-09-10

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- sysdeps/i386/fpu/bits/mathinline.h~	2003-08-26 17:28:53.000000000 -0700
+++ sysdeps/i386/fpu/bits/mathinline.h	2003-08-26 17:31:12.000000000 -0700
@@ -381,7 +381,7 @@
     ("fscale			# 2^int(x * log2(e))\n\t"		      \
      : "=t" (__temp) : "0" (1.0), "u" (__exponent));			      \
   __temp -= 1.0;							      \
-  return __temp + __value ?: __x
+  return __temp + __value ? __temp + __value : __x
 __inline_mathcodeNP_ (long double, __expm1l, __x, __expm1_code)
 
 
